arbor/cable_cell_param.cpp

```diff
@@ -68,4 +70,92 @@ cable_cell_parameter_set neuron_parameter_defaults = {
     },
 };
 
+
+std::vector<defaultable> cable_cell_parameter_set::serialize() const {
+    std::vector<defaultable> D;
+    if (init_membrane_potential) {
+        D.push_back(arb::init_membrane_potential{*this->init_membrane_potential});
+    }
+    if (temperature_K) {
+        D.push_back(arb::temperature_K{*this->temperature_K});
+    }
+    if (axial_resistivity) {
+        D.push_back(arb::axial_resistivity{*this->axial_resistivity});
+    }
+    if (membrane_capacitance) {
+        D.push_back(arb::membrane_capacitance{*this->membrane_capacitance});
+    }
+
+    //std::unordered_map<std::string, cable_cell_ion_data> ion_data;
+    for (auto& ion: ion_data) {
```

Comment:
How about
```
for (const auto& [name, data]: ion_data) {
   ...
}
```
?

Reply:
Good point. I am still getting used to using the C++17 syntactic sugar.

Reply:
I reverted this change, because it triggers a bug in Clang (compiler error).